client/my-sites/email/email-providers-stacked-comparison/provider-cards/utils.tsx

```diff
@@ -60,3 +86,74 @@ export const recordTracksEventAddToCartClick = (
 		user_cannot_add_email_code: userCannotAddEmailReason ? userCannotAddEmailReason.code : '',
 	} );
 };
+
+type PriceWithIntervalProps = {
+	className: string;
+	cost: number;
+	currencyCode: string;
+	hasDiscount: boolean;
+	sale?: number | null;
+	intervalLength: IntervalLength;
+};
+
+export const PriceWithInterval: FunctionComponent< PriceWithIntervalProps > = ( props ) => {
+	const { className, cost, currencyCode, hasDiscount, sale, intervalLength } = props;
+
+	const costClassName = `${ className }__${ hasDiscount ? 'discounted-price' : 'keep-main-price' }`;
+	const intervalLengthClassName = `${ className }__interval`;
+	const mailboxClassName = `${ className }__mailbox`;
+	const saleClassName = `${ className }__sale`;
+
+	const showSale = sale && sale !== 0;
+
+	const lineThrough = {
+		textDecoration: hasDiscount || showSale ? 'line-through' : 'solid ',
+	};
+
+	const priceSpan = (
+		<span style={ lineThrough } className={ costClassName }>
+			{ formatCurrency( cost ?? 0, currencyCode ) }
+		</span>
+	);
+	const saleSpan = (
+		<span className={ saleClassName }>{ formatCurrency( sale ?? 0, currencyCode ) }</span>
+	);
+	const intervalLengthSpan =
+		intervalLength === 'annually' ? (
+			<span className={ intervalLengthClassName }>/{ translate( 'annually' ) }</span>
+		) : (
+			<span className={ intervalLengthClassName }>/{ translate( 'monthly' ) }</span>
+		);
+
+	const mailboxSpan = <span className={ mailboxClassName }> /{ translate( 'mailbox' ) } </span>;
+
+	return (
+		<>
+			{ priceSpan } { showSale && saleSpan } { mailboxSpan } { intervalLengthSpan }
+		</>
+	);
+};
+
+type PriceBadgeProps = {
+	additionalPriceInformationComponent?: ReactElement | null;
+	className: string;
+	priceComponent: ReactElement;
+};
+
+export const PriceBadge: FunctionComponent< PriceBadgeProps > = ( props ) => {
+	const { additionalPriceInformationComponent, className, priceComponent } = props;
+	const priceBadgeClass = `${ className }__price-badge`;
+	const additionalPriceInformationClass = `${ className }__provider-additional-price-information`;
+	return (
+		<div className={ priceBadgeClass }>
+			<PromoCardPrice
+				formattedPrice={ priceComponent }
+				additionalPriceInformation={
+					<span className={ additionalPriceInformationClass }>
+						{ additionalPriceInformationComponent }
+					</span>
+				}
+			/>
+		</div>
+	);
+};
```

Comment:
These components already exist in `calypso/my-sites/email/email-providers-stacked-comparison/provider-cards/price-badge` and `calypso/my-sites/email/email-providers-stacked-comparison/provider-cards/price-with-interval`. We should not be adding them again.

TBH, I am not sure we need any changes to this file in this PR - they don't seem related to CSS.